core/trie/trie.go

```diff
@@ -43,27 +65,38 @@ type Trie struct {
 	hash    hashFunc
 }
 
-func NewTriePedersen(storage Storage, height uint, rootKey *bitset.BitSet) *Trie {
+func NewTriePedersen(storage Storage, height uint, rootKey *bitset.BitSet) (*Trie, error) {
 	return newTrie(storage, height, rootKey, crypto.Pedersen)
 }
 
-func NewTriePoseidon(storage Storage, height uint, rootKey *bitset.BitSet) *Trie {
+func NewTriePoseidon(storage Storage, height uint, rootKey *bitset.BitSet) (*Trie, error) {
 	return newTrie(storage, height, rootKey, crypto.Poseidon)
 }
 
-func newTrie(storage Storage, height uint, rootKey *bitset.BitSet, hash hashFunc) *Trie {
-	// Todo: set max height to 251 and set max key value accordingly
+func newTrie(storage Storage, height uint, rootKey *bitset.BitSet, hash hashFunc) (*Trie, error) {
+	if height > maxHeight {
+		return nil, &errMaxHeight{height}
+	}
+
+	// maxKey is 2^maxHeight - 1
+	maxKey = new(felt.Felt).Exp(new(felt.Felt).SetUint64(2), new(big.Int).SetUint64(uint64(maxHeight)))
```

Comment:
this should be a member of `Trie`, if not this applies for all tries of all height